src/SYNFPAN.m

```diff
@@ -0,0 +1,513 @@
+ . ;
+ . ; determine Panel type, code, coding system, and display text
+ . ;
+ . new paneltype set paneltype=$get(@json@("entry",SYNZI,"resource","code","text"))
+ . if paneltype="" set paneltype=$get(@json@("entry",SYNZI,"resource","code","coding",1,"display"))
+ . ;do log(jlog,"Panel type is: "_paneltype)
+ . do log(jlog,"Panel type is: "_obsdisplay)
+ . set @eval@("panels",SYNZI,"vars","type")=obsdisplay
+ . ;
+ . ; determine the id of the resource
+ . ;
+ . new id set id=$get(@json@("entry",SYNZI,"resource","id"))
+ . set @eval@("panels",SYNZI,"vars","id")=id
+ . d log(jlog,"ID is: "_id)
+ . ;
+ . ;Here's the spec for uploading a panel:
+ . ;KBANTEST ;
+ . ;N %,SAM,RC
+ . ;S SAM("LAB_PANEL")="CHEM 7"
+ . ;S SAM("LAB_TEST","BUN")=10
+ . ;S SAM("LAB_TEST","CO2")=34
+ . ;S SAM("LAB_TEST","GLUCOSE")=180
+ . ;S SAM("PAT_SSN")="999504449"
+ . ;S SAM("RESULT_DT")="NOW"
+ . ;S SAM("LOCATION")="3E NORTH"
+ . ;S SAM("COLLECTION_SAMPLE")="BLOOD" ; optional
+ . ;S %=$$LAB^ISIIMP12(.RC,.SAM)
+ . ;ZWRITE RC
+ . ;QUIT
+ . ;
+ . ; starting the parameter array with the panel level elements
+ . ;
+ . n MISC ; parameter array
+ . ;
+ . ; lab panel
+ . ;
+ . N PANEL
+ . S PANEL=$$MAP^SYNQLDM(loinc,"vistapanel")
+ . d log(jlog,"VistA panel is: "_PANEL)
+ . S MISC("LAB_PANEL")=PANEL
+ . ;
+ . ; patient
+ . ;
+ . s MISC("PAT_SSN")=$$GET1^DIQ(2,dfn_",","SSN")
+ . ;
+ . ; result date/time
+ . ;
+ . new effdate set effdate=$get(@json@("entry",SYNZI,"resource","effectiveDateTime"))
+ . do log(jlog,"effectiveDateTime is: "_effdate)
+ . set @eval@("panels",SYNZI,"vars","effectiveDateTime")=effdate
+ . new fmtime s fmtime=$$fhirTfm^SYNFUTL(effdate)
+ . ; time adjustment to avoid duplicates
+ . ;i $d(FMTIME(fmtime)) d  ;
+ . ;. ;ZWR FMTIME
+ . ;. n i s i=fmtime
+ . ;. f fmtime=i:.000001 q:'$d(FMTIME(fmtime))
+ . ;s FMTIME(fmtime)=""
+ . d log(jlog,"fileman dateTime is: "_fmtime)
+ . set @eval@("panels",SYNZI,"vars","fmDateTime")=fmtime ;
+ . new hl7time s hl7time=$$fhirThl7^SYNFUTL(effdate)
+ . d log(jlog,"hl7 dateTime is: "_hl7time)
+ . set @eval@("panels",SYNZI,"vars","hl7DateTime")=hl7time ;
+ . s MISC("RESULT_DT")=fmtime
+ . ;
+ . ; location
+ . ;
+ . s DHPLOC=$$MAP^SYNQLDM("OP","location")
+ . n DHPLOCIEN s DHPLOCIEN=$o(^SC("B",DHPLOC,""))
+ . if DHPLOCIEN="" S DHPLOCIEN=4
+ . ;s @eval@("labs",SYNZI,"parms","DHPLOC")=DHPLOC
+ . d log(jlog,"Location for outpatient is: #"_DHPLOCIEN_" "_DHPLOC)
+ . s MISC("LOCATION")=DHPLOC
+ . ;
+ . ; collection sample
+ . ;
+ . n CSAMP
+ . S CSAMP=$$GET1^DIQ(95.3,$p(loinc,"-"),4)
+ . I CSAMP["SER/PLAS" S CSAMP="SERUM"
+ . I CSAMP["Whole blood" S CSAMP="BLOOD"
+ . d log(jlog,"Collection sample is: "_CSAMP)
+ . s MISC("COLLECTION_SAMPLE")=CSAMP
+ . ;
+ . ;  ; add code to process DiagnosticReport results here
+ . ;
+ . n triples s triples=$na(@root@(ien))
+ . n atomptr s atomptr=$na(@json@("entry",SYNZI,"resource","result"))
+ . n rien s rien=""
+ . n zj s zj=0
+ . f  s zj=$o(@atomptr@(zj)) q:+zj=0  d  ;
```

Comment:
How do you check if a lab is a member of the VistA panel? I know you do that, but I can't see where.

Reply:
if response is 1^Included in Lab Panel #rien
and loadstatus is loaded

Reply:
Hmm... I don't think that's what I meant...

I mean, how do you know that PDW is in VistA but should be part of CBC but actually isn't part?